src/store/modules/wallet.js

```diff
@@ -46,8 +64,26 @@ const mutations = {
   [WALLET_SET_SIGNER](state, signer) {
     state.signer = signer;
   },
-  [types.WALLET_SET_LOGIN_ADDRESS](state, loginAddress) {
-    state.loginAddress = loginAddress;
+  [WALLET_SET_IS_LOGGING_IN](state, isLoggingIn) {
+    state.isLoggingIn = isLoggingIn;
+  },
+  [WALLET_SET_USER_INFO](state, userInfo) {
+    if (userInfo) {
+      const { user, email, emailUnconfirmed } = userInfo;
+      if (user !== undefined) {
+        state.loginAddress = user;
+      }
+      if (email !== undefined) {
+        state.email = email;
+      }
+      if (emailUnconfirmed !== undefined) {
+        state.emailUnverified = emailUnconfirmed;
+      }
+    } else {
+      state.loginAddress = '';
+      state.email = '';
+      state.emailUnverified = '';
+    }
```

Comment:
Not sure what are the `if`s for, what happen if we set these as `undefined` without checking?

Reply:
oh so it's used for updating individual field...
not sure if we will hit a case where a field is really not set for a user in db
can we split set and update as 2 mutations? set is used when we switch user, update is used for individual fields